connection_provider_pgx.go

```diff
@@ -11,7 +11,7 @@ import (
 // PgxConnectionProvider implements ConnectionProvider using pgx driver with connection pooling.
 type PgxConnectionProvider struct {
 	connectionStringFunc func(string) string
-	poolConfig           *pgxpool.Config
+	poolConfig           pgxpool.Config
```

Comment:
Pooling is a popular feature of the `pgx` library, so we will not miss much by declaring the memory for the struct straight away. At the same time, we get rid of many `if` statements, improving maintenance and time performance.